src/main/java/org/folio/rest/impl/base/AbstractEmail.java

```diff
@@ -183,17 +172,26 @@ protected Future<Collection<EmailEntity>> handleFailure(Collection<EmailEntity>
       .compose(r -> failedFuture(throwable));
   }
 
-  protected Future<Configurations> lookupConfiguration(Map<String, String> requestHeaders) {
-    MultiMap headers = MultiMap.caseInsensitiveMultiMap().addAll(requestHeaders);
-    String okapiUrl = headers.get(OKAPI_URL_HEADER);
-    String okapiToken = headers.get(OKAPI_HEADER_TOKEN);
-    String url = String.format(REQUEST_URL_TEMPLATE, okapiUrl, REQUEST_URI_PATH, MODULE_EMAIL_SMTP_SERVER);
-
-    return httpClient.getAbs(url)
-      .putHeader(OKAPI_HEADER_TOKEN, okapiToken)
-      .putHeader(OKAPI_HEADER_TENANT, tenantId)
-      .putHeader(CONTENT_TYPE, APPLICATION_JSON)
-      .putHeader(ACCEPT, APPLICATION_JSON)
+  private Future<SmtpConfiguration> lookupSmtpConfiguration(Map<String, String> requestHeaders) {
+    return smtpConfigurationService.getSmtpConfiguration()
+      .compose(EmailUtils::validateSmtpConfiguration)
+      .recover(throwable -> moveConfigsFromModConfigurationToLocalDb(requestHeaders));
+  }
+
+  private Future<SmtpConfiguration> moveConfigsFromModConfigurationToLocalDb(
+    Map<String, String> requestHeaders) {
+
```

Comment:
We can instantiate OkapiClient here and pass it into both methods below instead of `requestHeaders`

Reply:
Done